### PR TITLE
Update cache sync cache.

### PR DESCRIPTION
**Pull request by AurelienFT**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-protocol-worker/src/handlers/block_handler/cache.rs [hunk 1]
```diff
@@ -47,6 +47,35 @@ impl BlockCache {
...
         }
     }
+
+    pub fn update_cache(
+        &mut self,
+        peers_connected: HashSet<PeerId>,
+        max_known_blocks_by_peer: u32,
+    ) {
+        let peers: Vec<PeerId> = self
+            .max_known_blocks_by_peer
```
Comment (modship) [inline massa-protocol-worker/src/handlers/block_handler/cache.rs:57]: max_known_blocks_by_peer is u32 no ?

Comment (AurelienFT) [reply massa-protocol-worker/src/handlers/block_handler/cache.rs]: Fixed. I will fix my IDE also he was like "everythig is fine"

Comment (modship) [reply massa-protocol-worker/src/handlers/block_handler/cache.rs]: me too since few days

Comment (AurelienFT) [reply massa-protocol-worker/src/handlers/block_handler/cache.rs]: Compilation ended on my PC I fixed everything should be ok now

**Review by modship — APPROVED** [review]

**Pull request merged**